### PR TITLE
dev-util/diffball: disable static library

### PR DESCRIPTION
**Pull request by Alessandro-Barbieri**
Package-Manager: Portage-2.3.76, Repoman-2.3.17
Signed-off-by: Alessandro Barbieri <lssndrbarbieri@gmail.com>

### CODE REVIEW

File: dev-util/diffball/diffball-1.0.1-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~hppa ~ia64 ~mips ~ppc ~sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos"
+IUSE="debug"
+
+RDEPEND=">=sys-libs/zlib-1.1.4
+	>=app-arch/bzip2-1.0.2
+	app-arch/xz-utils"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig"
```
Comment (ConiKost) [inline dev-util/diffball/diffball-1.0.1-r1.ebuild]: With EAPI=7, ```virtual/pkgconfig``` is ```BDEPEND```

File: dev-util/diffball/diffball-1.0.1-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+EAPI=7
+
+inherit flag-o-matic
+
+DESCRIPTION="Delta compression suite for using/generating binary patches"
+HOMEPAGE="https://github.com/rafaelmartins/diffball"
+SRC_URI="https://storage.googleapis.com/google-code-archive-downloads/v2/code.google.com/${PN}/${P}.tar.bz2"
+
+LICENSE="BSD"
```
Comment (ConiKost) [inline dev-util/diffball/diffball-1.0.1-r1.ebuild]: According to source code, it's a different license?

File: dev-util/diffball/diffball-1.0.1-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+RESTRICT="strip"
+
+src_prepare() {
+	# fix bug 548316 by restoring pre-GCC5 inline semantics
+	append-cflags -std=gnu89
+	default
+}
+
+src_configure() {
+	econf \
```
Comment (ConiKost) [inline dev-util/diffball/diffball-1.0.1-r1.ebuild]: Do make it more elegant without breaking lines with slash, you could make an array.

File: dev-util/diffball/diffball-1.0.1-r1.ebuild (hunk 2 above)
Comment (Alessandro-Barbieri) [reply dev-util/diffball/diffball-1.0.1-r1.ebuild]: It's BSD 3 clauses
https://github.com/rafaelmartins/diffball/commit/95c18477dea9c9dd2b362ff40e5a7762195e8d98#diff-7116ef0705885343c9e1b2171a06be0e

Comment (ConiKost) [reply dev-util/diffball/diffball-1.0.1-r1.ebuild]: Source code indicates ```GPL-2+```. Source code overrides ```COPYING```.
See https://devmanual.gentoo.org/general-concepts/licenses/index.html.

File: dev-util/diffball/diffball-1.0.1-r1.ebuild (hunk 1 above)
Comment (SoapGentoo) [reply dev-util/diffball/diffball-1.0.1-r1.ebuild]: Don't include `RDEPEND` in `BDEPEND`

File: dev-util/diffball/diffball-1.0.1-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	default
+}
+
+src_configure() {
+	local myconf=(	--disable-static
+			$(use_enable debug asserts) )
+	econf "${myconf[@]}"
+}
+
+src_install(){
```
Comment (SoapGentoo) [inline dev-util/diffball/diffball-1.0.1-r1.ebuild]: `src_install() {`

File: dev-util/diffball/diffball-1.0.1-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+src_prepare() {
+	# fix bug 548316 by restoring pre-GCC5 inline semantics
+	append-cflags -std=gnu89
+	default
+}
+
+src_configure() {
+	local myconf=(	--disable-static
+			$(use_enable debug asserts) )
+	econf "${myconf[@]}"
```
Comment (SoapGentoo) [inline dev-util/diffball/diffball-1.0.1-r1.ebuild]: just inline this
```
econf --disable-static $(use_enable debug asserts)
```

File: dev-util/diffball/diffball-1.0.1-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+	app-arch/xz-utils"
+BDEPEND="${RDEPEND}
+	virtual/pkgconfig"
+
+# Invalid RESTRICT for source package. Investigate.
+RESTRICT="strip"
+
+src_prepare() {
+	# fix bug 548316 by restoring pre-GCC5 inline semantics
+	append-cflags -std=gnu89
```
Comment (SoapGentoo) [inline dev-util/diffball/diffball-1.0.1-r1.ebuild]: do this before the call to `econf` in `src_configure` and then nix `default`

File: dev-util/diffball/diffball-1.0.1-r1.ebuild [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+IUSE="debug"
+
+RDEPEND=">=sys-libs/zlib-1.1.4
+	>=app-arch/bzip2-1.0.2
+	app-arch/xz-utils"
+BDEPEND="${RDEPEND}
+	virtual/pkgconfig"
+
+# Invalid RESTRICT for source package. Investigate.
+RESTRICT="strip"
```
Comment (SoapGentoo) [inline dev-util/diffball/diffball-1.0.1-r1.ebuild]: put this after `IUSE`

**Comment by rafaelmartins** [comment]
as the libraries are public, you can't just disable static libraries. you should at least add `static-libs` use flag

**Comment by Alessandro-Barbieri** [comment]
@rafaelmartins we are getting rid of the `static-libs` useflag
see https://flameeyes.blog/2011/08/29/useless-flag-static-libs/
and https://archives.gentoo.org/gentoo-dev/message/2dada80c2b9c85b0e83e6328428bf8ab
for more context

**Comment by rafaelmartins** [comment]
> @rafaelmartins we are getting rid of the `static-libs` useflag
> see https://flameeyes.blog/2011/08/29/useless-flag-static-libs/
> and https://archives.gentoo.org/gentoo-dev/message/2dada80c2b9c85b0e83e6328428bf8ab
> for more context

this is ongoing for years, and the package is fine as is, either use the flag, or keep it as is. I'm not going to merge `--disable-static`

**Comment by Alessandro-Barbieri** [comment]
I'll leave the final word to @SoapGentoo

**Comment by rafaelmartins** [comment]
> I'll leave the final word to @SoapGentoo

as much as I value his input, he is not the maintainer, and there's no council decision about banning `static-libs` use flag afaik, so the ultimate word is still mine here.

**Comment by SoapGentoo** [comment]
https://flameeyes.blog/2011/08/29/useless-flag-static-libs/ - "This means that unless there are reverse dependencies building against the static libraries, the package should not have the static-libs USE flag at all."
But ok, we can devise a QA policy if you prefer.

**Comment by rafaelmartins** [comment]
> https://flameeyes.blog/2011/08/29/useless-flag-static-libs/ - "This means that unless there are reverse dependencies building against the static libraries, the package should not have the static-libs USE flag at all."
> But ok, we can devise a QA policy if you prefer.

as much as I respect Diego, his blog is not official gentoo qa policy. if you have some official policy to point me at, we can discuss what to do.

File: dev-util/diffball/diffball-1.0.1-r1.ebuild (hunk 1 above)
Comment (Alessandro-Barbieri) [reply dev-util/diffball/diffball-1.0.1-r1.ebuild]: done

**Pull request closed without merging**